### ISSUE TITLE
Multiple requests per template

### ISSUE THREAD

**Issue by NkxxkN** [projectdiscovery/nuclei#57]
## Multiple requests per template

Following up on first attempt of dirty fix: https://github.com/projectdiscovery/nuclei/pull/35

This is a regression introduced by https://github.com/projectdiscovery/nuclei/commit/2c58a3391329d200a9b06df80daf8542d92279f7. It used to work even though it had not been really heavily used looking at the variety of open source templates at the moment. 

A good use case for this is CORS check: https://github.com/projectdiscovery/nuclei-templates/pull/49. 

@Ice3man543 could you elaborate a bit on what you mean by "supporting chained requests is something that will break how things are done the way it's implemented" ?

**Comment by Ice3man543** [comment]
Hey @NkxxkN, so the way requests used to work in the past is they were sent one after the other. However, after adding per-request redirects we needed to create http clients per request. But following the current model of sending one request after the other per URL would result in creation of many http client objects for each URL. Hence we decided to create a client a request, complete all tasks and then do the next request.

However, we are fully open to any suggestions. I would love to implement the best way to support chained requests without breaking current functionality.
